### DISCUSSION TITLE
Unable to show the file name instead of file title inside the Search Result

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [microsoft-search/pnp-modern-search#2793] [Q&A]
I have added a PnP Search Result web part and i define the following settings:-

![image](https://user-images.githubusercontent.com/79726511/220453501-c0c2ef10-358d-4cbf-a229-b52a1d53d05b.png)

and i define the first slot to show the document name instead of title:-

![image](https://user-images.githubusercontent.com/79726511/220454274-6da4d603-1d1a-4761-9d5d-87708414826b.png)

and i defined the following Selected fields : -

![image](https://user-images.githubusercontent.com/79726511/220454546-3db7e42a-d407-4834-a5b3-4f1969fd0cc6.png)

but on the result >> i will get nothing inside the FileName:-

![image](https://user-images.githubusercontent.com/79726511/220454687-d7d95000-cd92-4231-af6a-054cdad69d24.png)

here is my display columns

![image](https://user-images.githubusercontent.com/79726511/220454823-e17a17b7-f17c-40e7-8185-ca506dcd2c27.png)

can anyone advice why i am not getting the file names?

Thanks

**Comment by kasperbolarsen** [comment]
It depends on what you mean by file names :-) 
If it is "name" or Title as below then you can use either the title or name property. 

![image](https://user-images.githubusercontent.com/20593570/220468214-1b4c2bf1-d111-4d86-bd02-75a88220049d.png)


I haven't had the time yet to update the PnP Modern Search documentation with a guide on how to use the Graph Explorer to identified the available properties, but it is on my to do list.

In the meantime, this might point you in the right direction:

![image](https://user-images.githubusercontent.com/20593570/220469087-d2a4b589-e1f7-43e7-ba6b-4c13e9cb1352.png)

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen thanks for the update. but still not working on my end. here what i have:-

LayoutSlots:-

![image](https://user-images.githubusercontent.com/79726511/220474744-f73a1b55-5286-4828-9365-a6e8c76f24d8.png)

and inside the Managed Columns:-

![image](https://user-images.githubusercontent.com/79726511/220475206-653c85ca-ed6a-4b5e-96a5-7ae2dd63e589.png)

but still the file name is empty ?? any advice please?

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen  Appreciate if you can further help with the above?
Thanks

**Reply by wobba** [reply]
Use the debug template and check the exact json path to the prop to validate that `resource.name` is correct.
